pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/StContainsFunction.java

```diff
@@ -83,9 +82,6 @@ public TransformResultMetadata getResultMetadata() {
     for (int i = 0; i < projectionBlock.getNumDocs(); i++) {
       Geometry firstGeometry = GeometrySerializer.deserialize(firstValues[i]);
       Geometry secondGeometry = GeometrySerializer.deserialize(secondValues[i]);
-      if (GeometryUtils.isGeography(firstGeometry) || GeometryUtils.isGeography(secondGeometry)) {
```

Comment:
why did you remove this? performance?

Reply:
this check is needed, the current implementation does not handle geography correctly.

Reply:
why does it not work for geography?